src/libLookup/Lookup.cpp

```diff
@@ -919,19 +919,35 @@ bool Lookup::ProcessGetDSInfoFromSeed(const bytes& message, unsigned int offset,
 void Lookup::SendMessageToRandomSeedNode(const bytes& message) const {
   LOG_MARKER();
 
-  lock_guard<mutex> lock(m_mutexSeedNodes);
-  if (0 == m_seedNodes.size()) {
-    LOG_GENERAL(WARNING, "Seed nodes are empty");
-    return;
+  VectorOfNode notBlackListedSeedNodes;
+  {
+    lock_guard<mutex> lock(m_mutexSeedNodes);
+    if (0 == m_seedNodes.size()) {
+      LOG_GENERAL(WARNING, "Seed nodes are empty");
+      return;
+    }
+
+    for (const auto& node : m_seedNodes) {
+      auto seedNodeIpToSend = TryGettingResolvedIP(node.second);
+      if (!Blacklist::GetInstance().Exist(seedNodeIpToSend)) {
+        notBlackListedSeedNodes.push_back(node);
```

Comment:
We can store the `resolvedIP`  as well.  So that we don't need to resolve IPS for all `notBlacklistedSeedNodes`again at line 947.